File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums):
        res = []
        n = 1
        for num in nums:
            res.append(n)
            n *= num
        m = nums[-1]
        x = -1
        for num in nums[::-1]:
            if x == -1:
                None
                x += -1
            else:
                res[x] *= m
                m *= num
                x += -1
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0, j = 1;
        int def = target - nums[i];
        while(i != j && def != nums[j]) {
            if(size_t(j) < nums.size() - 1) {
                j++;
            } else if(size_t(i) < nums.size() - 2) {
                i++;
                j = i + 1;
                def = target - nums[i];
            } else 
                break;
        }

        vector<int> res = {i, j};
        return res;
    }
};
